### ISSUE TITLE
False negative when using a TypeVar bound to a TypedDict

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#5708]
The following code should produce a type error.

```python
from typing import Generic, TypeVar, TypedDict

class TD1(TypedDict):
    a: str

T = TypeVar("T", bound=TD1)

class G(Generic[T]):
    def method1(self) -> T:
        return {"a": ""} # No error!
```

This is related to mypy issue https://github.com/python/mypy/issues/11812.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.323, which I just published. It will also be included in a future release of pylance.
